### PR TITLE
READY: Correctly process Riak generated keys

### LINKED ISSUE #287: incorrect response on put request when using autogenerated keys [JIRA: CLIENTS-874]
Hi

I'm running a Riak 2.0.6-1 cluster with riak erlang client lib version 2.1.1, riak_pb 2.1.0.2.

According to the README.md

> If the return value of the last command was anything but the atom ok (or {ok, Key} when you instruct the server to generate the key), then the store failed. The return value may give you a clue as to why the store failed...

Which is expressed in the code[1] as well. This is the same function header as in 2.1.1.

While the response I get is:

```
{ok,{riakc_obj,{<<"bucket_type">>,<<"bucket">>},
               <<"DYih3nkXtFi6ytGm7T2q7pjG2DG">>,undefined,[],undefined,
               undefined}}
```

The object itself is created with undefined as key:
`              Obj = riakc_obj:new(?REQUEST_BUCKET, undefined, to_binary(Req)),`

When I traced the process_response/3 function I saw this, I'll only include the interesting bits:
```
(<0.231.0>) call riakc_pb_socket:process_response(
                   {request,#Ref<0.0.2.19281>,
                      {rpbputreq, ... },
                   {rpbputresp,[],undefined,<<"1lZ4hsH4Y6RB5aYtkVj8JTqNCAh">>},
                   {state, ...}
                  )
```

rpbputresp is defined as:
```
-record(rpbputresp, {
    content = [],
    vclock,
    key
}).
```

So if we review the code call, we see that content is set to [], while in [1] it's expected to be undefined thus the pattern match doesn't work.

I changed the code to pattern match against content [] and I did receive {ok, <<"key">>}. 

In [2] for #rpbgetresp [] is being pattern matched not undefined.

I can create a PR to correct this if this is indeed a bug. If it isn't I'm curious to know what's wrong with my setup.
Cheers,

[1] https://github.com/basho/riak-erlang-client/blob/eed70717ece9c6796e29b5a326fa6af98b2ec43a/src/riakc_pb_socket.erl#L1661-L1666
[2] https://github.com/basho/riak-erlang-client/blob/eed70717ece9c6796e29b5a326fa6af98b2ec43a/src/riakc_pb_socket.erl#L1644-L1645

### PR DESCRIPTION
**Pull request by lukebakken**
@sata ping

Fixes #287 

* Modify test to test that returned object from Riak is actually a binary

### CODE REVIEW

**Comment by lukebakken** [comment]
@javajolt - the first commit changes the test to demonstrate the issue, the second fixes it.

Thank you @sata

File: src/riakc_pb_socket.erl [hunk 1]
```diff
@@ -3286,8 +3300,8 @@ live_node_tests() ->
                     {ok, Obj1} = ?MODULE:put(Pid, PO, [return_body]),
                     {ok, Obj2} = ?MODULE:put(Pid, PO, [return_body]),
                     %% Make sure the same key isn't generated twice
-                    ?assertEqual(element(1, Obj1), riakc_obj),
-                    ?assertEqual(element(1, Obj2), riakc_obj),
+                    ?assertEqual(riakc_obj, element(1, Obj1)),
+                    ?assertEqual(riakc_obj, element(1, Obj2)),
```
Comment (lukebakken) [inline src/riakc_pb_socket.erl]: Docs say the order should be switched:

http://erlang.org/doc/apps/eunit/chapter.html#Assert_macros

**Comment by javajolt** [comment]
@lukebakken I tried testing with just the first commit and the test passes.  I was expecting it to fail without your fix.  Am I mistaken?

**Comment by javajolt** [comment]
Live tests confirm the fix.

**Comment by javajolt** [comment]
:+1: d75ee0f

**Pull request merged**